pkg/brokerapi/service.go

```diff
@@ -21,14 +21,24 @@ package brokerapi
 //
 // https://github.com/openservicebrokerapi/servicebroker/blob/v2.12/spec.md#service-objects
 type Service struct {
-	Name            string        `json:"name"`
-	ID              string        `json:"id"`
-	Description     string        `json:"description"`
-	Tags            []string      `json:"tags,omitempty"`
-	Requires        []string      `json:"requires,omitempty"`
-	Bindable        bool          `json:"bindable"`
-	Metadata        interface{}   `json:"metadata,omitempty"`
-	DashboardClient interface{}   `json:"dashboard_client"`
-	PlanUpdateable  bool          `json:"plan_updateable,omitempty"`
-	Plans           []ServicePlan `json:"plans"`
+	Name            string          `json:"name"`
+	ID              string          `json:"id"`
+	Description     string          `json:"description"`
+	Tags            []string        `json:"tags,omitempty"`
+	Requires        []string        `json:"requires,omitempty"`
+	Bindable        bool            `json:"bindable"`
+	Metadata        interface{}     `json:"metadata,omitempty"`
+	DashboardClient DashboardClient `json:"dashboard_client"`
+	PlanUpdateable  bool            `json:"plan_updateable,omitempty"`
+	Plans           []ServicePlan   `json:"plans"`
+}
+
+// DashboardClient represents a dashboard_client which contains the data necessary to activate
+// the Dashboard SSO feature for this service.
+//
+// https://github.com/openservicebrokerapi/servicebroker/blob/v2.12/spec.md#dashboard-client-object
+type DashboardClient struct {
+	ID          string `json:"id,omitempty"`
+	Secret      string `json:"secret,omitempty"`
+	RedirectURI string `json:"redirect_uri,omitempty"`
```

Comment:
The downsize to this is that we lose any extension properties that might have been in there.  The spec doesn't say whether there could be any or not though - we should get that fixed - but my preference would be to support them.  Is there a way to not lose them but still have these 3 be well defined?

Reply:
what about add one more field for extension ?
   Config interface{} `json: "config,omitempty"`

Reply:
for now we can just keep this as-is and wait for the spec to decide what they want to do.

Reply:
+1 on what @duglin said here. I vote for waiting until there's more clarity in the spec